### DISCUSSION TITLE
Inconsistent commit time for `tycho-buildtimestamp-jgit`

### DISCUSSION THREAD

**Discussion by EhabY** [eclipse-tycho/tycho#2567] [Q&A]
Hi,

How can you generate the build qualifier to a file? This is very useful for our CI, and we'd like to do this using maven, instead of using regex's after the fact. We've tried `tycho-buildtimestamp-jgit`, but it bases the qualifier on the latest commit in that module, not the root git repository. We're currently in a situation where we use `-DforceContextQualifier` to always use the same qualifier on all modules.

Alternatively. is there a way to make it such that all modules use the same (latest) commit time?

**Comment by laeubi** [comment]
> We've tried tycho-buildtimestamp-jgit, but it bases the qualifier on the latest commit in that module, not the root git repository

That is actually the purpose, why do you think it does not suffice? Please note that technically a build can consist of multiple repository checkouts so it is not always possible to determine the "root", so usually the last change of the module gives a much better/stable qualifier.

**Reply by EhabY** [reply]
We basically want the "shared" qualifier to be the same one as in `git show -s --format=%ci HEAD`. We use a single git repository so it's expected that they all have the same git commit time

**Reply by laeubi** [reply]
`-DforceContextQualifier` seems a good choice then, can't you format the output of `git-show` the desired way and pass this this in your cibuild?

**Reply by EhabY** [reply]
Indeed that's what we are doing now - we thought we could have done it from within the maven build

Anyway, thank you!

**Reply by Bananeweizen** [reply]
If you want to unify everything, then it would only be plausible to increase every version number via tycho-versions-plugin:set-version as the last commit before building a release. That in turn would ensure all modules have the same timestamp, because the version number change would affect every module.
